packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { PropTypes } from 'react';
+import moment from 'moment';
+import TextMask from 'react-text-mask';
+import TimePipe from './TimePipe';
+import './TimeInput.scss';
+
+const propTypes = {
+  /**
+   * A moment object to use as the default initial time.
+   */
+  defaultTime: PropTypes.object,
+    /**
+   * A callback function to execute when a time value is entered.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  defaultTime: null,
+  onChange: null,
+};
+
+class TimeInput extends React.Component {
+  static userLocale() {
+    // TODO: Get the locale from i18nProvider
+    return 'en-US';
+  }
+
+  static timeFormat() {
+    // TODO: Get the time format translation.
+    return 'HH:mm';
+  }
+
+  static formattedTime(timeMoment) {
+    if (timeMoment) {
+      return timeMoment.clone().locale(TimeInput.userLocale()).format(TimeInput.timeFormat());
+    }
+
+    return '';
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      defaultTime: TimeInput.formattedTime(this.props.defaultTime),
+      value: TimeInput.formattedTime(this.props.defaultTime),
```

Comment:
Yup, dropped 'this'. https://github.com/cerner/terra-core/commit/eefcee225cc9d65e5c1891c4e3958eea9382a8a5